bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/ImageProcessingDescriptor.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.swt.internal.image;
+
+import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.*;
+
+public interface ImageProcessingDescriptor {
+
+	RGBA adaptPixelValue(int red, int green, int blue, int alpha);
+
+	public static ImageProcessingDescriptor createHSBImageProcessingDescriptor(float hueFactor, float saturationFactor,
```

Comment:
Just a nit-pick, but isn't mentioning the ImageProcessingDescriptor again in the method name somehow duplicated, if one assumes that this called with the interface as qualifier?
`ImageProcessingDescriptor.createHSBImageProcessingDescriptor`

If we make the method shorter it would also highlight the 'type' of transformation better, for example:
`ImageProcessingDescriptor.createHSB` `ImageProcessingDescriptor.forHSB` or `ImageProcessingDescriptor.ofHSB` or alike.

Reply:
Also thought about that and then went for the non-qualified way, but I agree that using the qualified way with those short factory method names is better to avoid duplication within the qualifier. I adapted the PR accordingly.